yamlstream.go

```diff
@@ -0,0 +1,64 @@
+package yaml
+
+import (
+	"io"
+	"reflect"
+)
+
+type Decoder struct {
+	d *decoder
+	p *parser
+}
+
+func (dec *Decoder) Decode(out interface{}) (err error) {
+	defer handleErr(&err)
+	if dec.p.event.typ == yaml_STREAM_END_EVENT {
+		return io.EOF
+	}
+	node := dec.p.parse()
+	if node != nil {
+		v := reflect.ValueOf(out)
+		if v.Kind() == reflect.Ptr && !v.IsNil() {
+			v = v.Elem()
+		}
+		dec.d.unmarshal(node, v)
+	}
+	if len(dec.d.terrors) > 0 {
+		return &TypeError{dec.d.terrors}
+	}
+	return nil
+}
+
+func (dec *Decoder) Close() {
```

Comment:
I don't think this method is needed - all the destroy method does is zero some memory. I think we can leave the garbage collector to do that.